### ISSUE TITLE
Lamda resolver does not work with Transformer V2 any more.  help needed

### ISSUE THREAD

**Issue by ErhardScampi** [aws-amplify/amplify-category-api#1397]
### How did you install the Amplify CLI?

amplify upgrade

### If applicable, what version of Node.js are you using?

v16.18.0

### Amplify CLI Version

11.0.3

### What operating system are you using?

MAC M1

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No changes

### Describe the bug

I created a lamda function according this description and called it .
https://aws.amazon.com/de/blogs/mobile/amplify-framework-adds-support-for-aws-lambda-functions-and-amazon-dynamodb-custom-indexes-in-graphql-schemas/
and
https://docs.amplify.aws/cli-legacy/graphql-transformer/function/

it worked in Transformer v1 but after updating to Transformer v2 it does not work any more.

Here the schema.graphql snipped :

type Query {
    myqueryresolver(params: String): String
    @function(name: "myqueryresolver-${env}")
    @auth(rules: [{ allow: public, provider: iam }, { allow: owner }])
}

Here the calling snipped 

console.log("-----Begin ----- > isAuthenticated");
    console.log(isAuthenticated);
    console.log("-----Ende ----- > isAuthenticated");
    const dataObj = await API.graphql({
      query: myqueryresolver,
      variables: { params: JSON.stringify(params) },
      authMode: isAuthenticated ? "AMAZON_COGNITO_USER_POOLS" : "AWS_IAM",
    });
    const data = JSON.parse(dataObj.data.myqueryresolver);
    
    
    And here the output

-----Begin ----- > isAuthenticated
cloud.js?f231:531 true
cloud.js?f231:532 -----Ende ----- > isAuthenticated
cloud.js?f231:541 Error
cloud.js?f231:542 {data: {…}, errors: Array(1)}data: {myqueryresolver: null}errors: Array(1)0: {path: Array(1), data: null, errorType: 'Unauthorized', errorInfo: null, locations: Array(1), …}length: 1[[Prototype]]: Array(0)[[Prototype]]: Object
runtime-core.esm-bundler.js?f781:40 [Vue warn]: Unh
    
    
The user is logged in other "normal" queries work. 

What to do ?    
In v1 it worked.
    
    

### Expected behavior

Working lamda resolver

### Reproduction steps

Create a lamda function
add a @function query to schema.graphql.
call it like described in the bug report.

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by ErhardScampi** [comment]
Hi @AnilMaktala can't do that I opend an issue on that 1312
Here my output on screen. As you see I want to cd in the report dir - the report dir does not exist (last two lines


daphne code$ amplify diagnose --send-report

Learn more at https://docs.amplify.aws/cli/reference/diagnose/

✅ Report saved: /var/folders/2w/cs5c31ln07l07km7mt594pnc0000gq/T/Artbarrack/report-1681405127294.zip

✔ Done

Project Identifier: c1b0a3da960647a9b8b73acc917781c5

daphne code$ cd /var/folders/2w/cs5c31ln07l07km7mt594pnc0000gq/T/A
AppTranslocation/         AudioComponentRegistrar/

**Comment by ErhardScampi** [comment]
@AnilMaktala you can say me the explicit files you need. I send you.
What I did I made an amplify update function.  to regenerate the files concerning the lamdafunctions but nothing changed.   Should I send the generated vtl files ?
My project was a migration from transformer v1.

**Comment by ErhardScampi** [comment]
I have found the error of #1312. So I can you send the file
[report-1681422466055.zip](https://github.com/aws-amplify/amplify-category-api/files/11226762/report-1681422466055.zip)

**Comment by ErhardScampi** [comment]
I struggle with transformer two.  Would You please look at the attached pdf where I put all the Information. Perhaps I make something completly wrong but I do not know what.   The goal is to gain crud access for logged in users and Readonly access for public users. 
[__ProblemTransformerV2.pdf](https://github.com/aws-amplify/amplify-category-api/files/11243114/__ProblemTransformerV2.pdf)

**Comment by AnilMaktala** [comment]
Hi @ErhardScampi, Thank you for providing additional information and will dig deep into this today.

**Comment by AnilMaktala** [comment]
Hey @ErhardScampi, Apologies for the delay.
It appears that the `ownerField` configuration is causing an issue, as the `owner` property is not found in the table properties. Since `owner` is a predefined keyword, could you please try renaming it to a different name, as shown below, and let me know if this resolves the issue?

```
type Todo @model @auth(rules: [{ allow: owner, ownerField: "author" }]) {
  content: String
  author: String # record owner information now stored in "author" field
}
```
Please refer to our [doc's](https://docs.amplify.aws/cli/graphql/authorization-rules/#per-user--owner-based-data-access) for more information.

**Comment by ErhardScampi** [comment]
Thanks for the hint but there is not other result - no access.  

So what I did :
I created a complete new App simply toDo with amplify init with no frontend. 
I used amplify studio to create a user and to add some data.
I fired some queries :
User logged in ok
API-key ok
IAM not ok.  -> should use the unauthRole ? But I get "Unauthorized"

Test it yourself -> then You see

This is the generated schema (I added a field „age“) :

type Todo @model @auth(rules: [{allow: public, operations: [read], provider: iam}, {allow: public, operations: [read]}, {allow: private}]) {
  id: ID!
  name: String!
  description: String
  age: Int
}


So I went back to my migrated application :

So I added API-key access to my migrated Applikation (from V1 to V2).
But no access via API KEY and via IAM. Even via API KEY is access denied.
All queries made by the Amplify studio console. The only access is user loggedIn.

My goal is to allow not logged In users to read all data (In the first step). The user should create and update and delete the items.
I have many in production in an unstable state now. So it is important for me to solve this problem somehow.  There is no consistent example out there with transformer 2 and public access which explains a working example.
